federationtypes.go

```diff
@@ -0,0 +1,97 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A SendRequest is the content of a request to PUT /_matrix/federation/v1/send/{txnID}/
+type SendRequest Transaction
+
+// A SendResponse is the content of a response to PUT /_matrix/federation/v1/send/{txnID}/
+type SendResponse struct {
+	// Map of string room event ID to the result of processing that event.
```

Comment:
> string room event ID

Er... what? I can tell it's a string already, but is this a room ID or event ID as the key? Or maybe it's a concatenation of both?